src/test/java/uk/gov/hmcts/reform/jobscheduler/jobs/HttpCallJobTest.java

```diff
@@ -49,22 +53,93 @@ public void execute_should_send_http_request_specified_in_job_details() {
         given(context.getJobDetail())
             .willReturn(newJob(HttpCallJob.class).withIdentity("id", "group").build());
 
-        given(actionExtractor.extract(context))
-            .willReturn(new HttpAction(
-                "http://localhost:8080/hello-world",
-                HttpMethod.POST,
-                ImmutableMap.of("ServiceAuthorization", "some-token"),
-                "some-body"
-            ));
+        given(authTokenGenerator.generate()).willReturn("newly-generated-token");
+    }
+
+    @Test
+    public void execute_calls_given_endpoint_url() {
+        //given
+        actionHadHeadersSetTo(Collections.emptyMap());
+
+        // when
+        executingHttpCallJob();
+
+        // then
+        verify(postRequestedFor(urlEqualTo("/hello-world")));
+    }
+
+    @Test
+    public void execute_calls_endpoint_with_given_body() {
+        //given
+        actionHadHeadersSetTo(Collections.emptyMap());
+
+        // when
+        executingHttpCallJob();
+
+        // then
+        verify(
+            postRequestedFor(urlEqualTo("/hello-world"))
+                .withRequestBody(equalTo("some-body"))
+        );
+    }
+
+    @Test
+    public void execute_adds_service_authorization_header() {
+        // given
+        actionHadHeadersSetTo(Collections.emptyMap());
 
         // when
-        new HttpCallJob(restTemplate, actionExtractor).execute(context);
+        executingHttpCallJob();
 
         // then
         verify(
             postRequestedFor(urlEqualTo("/hello-world"))
-                .withHeader("ServiceAuthorization", equalTo("some-token"))
+                .withHeader("ServiceAuthorization", equalTo("newly-generated-token"))
+        );
+    }
+
+    @Test
+    public void execute_replaces_existing_service_authorization_header() {
+        // given
+        actionHadHeadersSetTo(ImmutableMap.of("ServiceAuthorization", "some-token"));
+
+        // when
+        executingHttpCallJob();
+
+        // then
+        verify(
+            postRequestedFor(urlEqualTo("/hello-world"))
+                .withHeader("ServiceAuthorization", equalTo("newly-generated-token"))
                 .withRequestBody(equalTo("some-body"))
         );
     }
+
+    @Test
+    public void execute_preserves_non_service_authorization_headers() {
+        // given
+        actionHadHeadersSetTo(ImmutableMap.of("X-Custom-Header", "anything"));
+
+        // when
+        executingHttpCallJob();
+
+        // then
+        verify(
+            postRequestedFor(urlEqualTo("/hello-world"))
+                .withHeader("X-Custom-Header", equalTo("anything"))
+        );
+    }
+
+    private void actionHadHeadersSetTo(Map<String, String> headers) {
```

Comment:
I see that the intention is to make given-when-then read like a proper sentence, but by convention method names should be imperative, i.e. `setActionHeadersTo`